mesonbuild/dependencies/cmake.py

```diff
@@ -637,7 +637,7 @@ def get_variable(self, *, cmake: T.Optional[str] = None, pkgconfig: T.Optional[s
                 if len(v) == 1:
                     return v[0]
                 elif v:
-                    return v
+                    raise DependencyException(f'Got cmake variable (always a list) for {self!r} with more than one value, violating design constraints')
```

Comment:
@mensinda I don't understand what cmake is doing here. Please advise. Can we restrict this?

Reply:
In CMake, everything is a string and lists are strings with `;` separators. Thus the trace parser automatically splits the strings in the CMake variables to make processing easier.

So storing lists with more than one element in a variable is possible in CMake. If you we want to access them with `get_variable()` we either have to support lists as a return type or  `';'.join(v)` them and let the user in the `meson.build` deal with it.

Reply:
It's fine if the traceparser wants to split them internally so that it can reference splitted strings when handling dependency targets.

I do not understand however, the use case of using `dep.get_variable()` on a splitted string.

Under what circumstances would a user actually try to do that from a meson.build?

Reply:
I think in that case trying to use `${MY_FILES}` would be a sandbox violation anyway.

Reply:
Fine, whatever :) we can return a joined string and let the user cope with it.